classes/order/Order.php

```diff
@@ -966,10 +966,11 @@ public static function getCustomerOrders($id_customer, $show_hidden_status = fal
 
         foreach ($res as $key => $val) {
             // In case order creation crashed midway some data might be absent
-            $orderState = !empty($val['id_order_state']) ? $indexedOrderStates[$val['id_order_state']] : null;
-            $res[$key]['order_state'] = $orderState['name'] ?: null;
-            $res[$key]['invoice'] = $orderState['invoice'] ?: null;
-            $res[$key]['order_state_color'] = $orderState['color'] ?: null;
+            // also use the null coalescing operator, in case our values are `null`
+            $orderState = !empty($val['id_order_state']) ? ($indexedOrderStates[$val['id_order_state']] ?? null) : null;
+            $res[$key]['order_state'] = $orderState['name'] ?? null ?: null;
+            $res[$key]['invoice'] = $orderState['invoice'] ?? null ?: null;
+            $res[$key]['order_state_color'] = $orderState['color'] ?? null ?: null;
```

Comment:
Instead of having a bunch of conditionals and null coalescence , it can be as simple as an initial check:
```suggestion
            if (empty($val['id_order_state']) || empty($indexedOrderStates[$val['id_order_state']])) {
                continue;
            }
            
            $orderState = $indexedOrderStates[$val['id_order_state']];
            $res[$key]['order_state'] = $orderState['name'];
            $res[$key]['invoice'] = $orderState['invoice'];
            $res[$key]['order_state_color'] = $orderState['color'];
```
or if we need both `['order_state']` and `['order_state_color']` to be defined:

```suggestion
            if (empty($val['id_order_state']) || empty($indexedOrderStates[$val['id_order_state']])) {
                $res[$key]['order_state'] = $res[$key]['order_state_color'] = null;
                continue;
            }
            
            $orderState = $indexedOrderStates[$val['id_order_state']];
            $res[$key]['order_state'] = $orderState['name'];
            $res[$key]['invoice'] = $orderState['invoice'];
            $res[$key]['order_state_color'] = $orderState['color'];
```

Reply:
Yes, you're right, @davidglezz. I applied your second suggestion. Thank you!